File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/util/Lang.java
package de.dimedis.mobileentry.util;

import android.content.Context;
import android.content.res.Configuration;
import android.text.TextUtils;

import java.util.Locale;

public class Lang {
    public static void setLocation(Context context, String language) {
        Locale locale = TextUtils.isEmpty(language) ? Locale.ROOT : new Locale(language);
        Locale.setDefault(locale);
        Configuration config = new Configuration();
        config.locale = locale;
        context.getApplicationContext().getResources().updateConfiguration(config, null);
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/util/Lang.kt
package de.dimedis.mobileentry.util

import android.content.Context
import android.content.res.Configuration
import android.text.TextUtils
import java.util.*

object Lang {
    fun setLocation(context: Context, language: String?) {
        val locale = if (TextUtils.isEmpty(language)) Locale.ROOT else language?.let { Locale(it) }
        if (locale != null) {
            Locale.setDefault(locale)
        }
        val config = Configuration()
        config.locale = locale
        context.applicationContext.resources.updateConfiguration(config, null)
    }
}